backend/battling/api/serializers.py

```diff
@@ -10,7 +11,71 @@ class Meta:
         fields = ("email", "id")
 
 
+class PokemonSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Pokemon
+        fields = ("poke_id", "name", "img_url", "attack", "defense", "hp")
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    pokemon_1 = PokemonSerializer()
+    pokemon_2 = PokemonSerializer()
+    pokemon_3 = PokemonSerializer()
+
+    class Meta:
+        model = PokemonTeam
+        fields = ("pokemon_1", "pokemon_2", "pokemon_3")
+
+
 class BattleSerializer(serializers.ModelSerializer):
     class Meta:
         model = Battle
         fields = ("id", "creator", "opponent", "winner")
+
+
+class BattleDetailSerializer(serializers.ModelSerializer):
+    creator = UserSerializer()
+    creator_team = serializers.SerializerMethodField()
+    opponent = UserSerializer()
+    opponent_team = serializers.SerializerMethodField()
+    winner = UserSerializer()
+
+    def get_creator_team(self, obj):
+        creator_team = Team.objects.filter(battle=obj, trainer=obj.creator).first()
+        creator_pokemon_list = PokemonTeam.objects.filter(team=creator_team).first()
+
+        if not creator_team:
+            return {}
+
+        creator_serializer = TeamSerializer(instance=creator_pokemon_list)
```

Comment:
This is quite ingenious. However, there's an easier way to do this: https://www.django-rest-framework.org/api-guide/relations/#nested-relationships

It's interesting that you are already using something similar with the `UserSerializer`. I recommend you to replace `get_creator_team` and `get_opponent_team` with this